clients/ui/frontend/src/app/pages/modelCatalog/utils/modelCatalogUtils.ts

```diff
@@ -118,32 +121,67 @@ export const shouldShowValidatedInsights = (
   artifacts: CatalogArtifacts[],
 ): boolean => isModelValidated(model) && hasPerformanceArtifacts(artifacts);
 
-const isStringFilterValid = <K extends ModelCatalogStringFilterKey>(
-  filterKey: K,
-  value: ModelCatalogStringFilterValueType[ModelCatalogStringFilterKey][],
-): value is ModelCatalogFilterStates[K] => isEnumMember(filterKey, ModelCatalogStringFilterKey);
+// Define array-based filter keys (excluding USE_CASE which is single-selection)
+type ArrayFilterKey =
+  | ModelCatalogStringFilterKey.TASK
+  | ModelCatalogStringFilterKey.PROVIDER
+  | ModelCatalogStringFilterKey.LICENSE
+  | ModelCatalogStringFilterKey.LANGUAGE
+  | ModelCatalogStringFilterKey.HARDWARE_TYPE;
+
+// Type mapping for array filter values
+type ArrayFilterValueType = {
+  [ModelCatalogStringFilterKey.TASK]: ModelCatalogTask;
+  [ModelCatalogStringFilterKey.PROVIDER]: ModelCatalogProvider;
+  [ModelCatalogStringFilterKey.LICENSE]: ModelCatalogLicense;
+  [ModelCatalogStringFilterKey.LANGUAGE]: AllLanguageCode;
+  [ModelCatalogStringFilterKey.HARDWARE_TYPE]: string;
+};
+
+// Type guard to check if a value is an array of the expected type
+const isArrayOfValues = <T>(value: unknown): value is T[] => Array.isArray(value);
+
+// Type guard to check if filter key is valid for array operations
+const isArrayFilterKey = (filterKey: string): filterKey is ArrayFilterKey =>
+  isEnumMember(filterKey, ModelCatalogStringFilterKey) &&
+  filterKey !== ModelCatalogStringFilterKey.USE_CASE;
```

Comment:
Well, we'll have to come back to this - we're presenting it as single selection with the current designs but it doesn't inherently have to be. When this gets lifted to the models page we'll want it to be multi-select I imagine, and it would match models that have perf artifacts for any of the selected use cases.

I think as far as the types go it's just another string filter like the others. We can likely revert some of these changes and some below, I'll follow up to take a closer look.